CONTRIBUTING.md

```diff
@@ -0,0 +1,261 @@
+```
+
+Similarly to unit tests, for finer control over which tests get executed, e.g. to run only 1 specific test case, execute:
+
+```shell
+tox -e integration -- tests/integration/test_package_managers.py::test_packages[gomod_without_deps]
+```
+
+### Running integration tests and generating new test data
+
+To re-generate new data (output, dependencies checksums, vendor checksums) and run integration tests with them:
+
+```shell
+make GENERATE_TEST_DATA=true test-integration
+```
+
+Generate data for test cases matching a pytest pattern:
+
+```shell
+CACHI2_GENERATE_TEST_DATA=true tox -e integration -- -k gomod
+```
+
+### Adding new dependencies to the project
+
+Sometimes when working on adding a new feature you may need to add a new dependency to the project.
+Usually, one commonly goes about it by adding the dependency to one of the ``requirements`` files
+or the more modern and standardized ``pyproject.toml`` file.
+In our case, dependencies must always be added to the ``pyproject.toml`` file as the
+``requirements`` files are generated by the ``pip-compile`` tool to not only pin versions of all
+dependencies but also to resolve and pin transitive dependencies. Since our ``pip-compile``
+environment is tied to Python 3.9, we have a Makefile target that runs the tool in a container
+image so you don't have to install another Python version locally just because of this. To
+re-generate the set of dependencies, run the following in the repository and commit the changes:
+
+```
+$ make pip-compile
+```
+
+## Releasing
+
+To release a new version of Cachi2, simply create a [GitHub release][cachi2-releases]. Note that
+Cachi2 follows [semantic versioning](https://semver.org/) rules.
+
+Upon release, the [.tekton/release.yaml](.tekton/release.yaml) pipeline tags the corresponding
+[Cachi2 image][cachi2-container] with the newly released version tag (after validating that the
+tag follows the expected format: `$major.$minor.$patch`, without a `v` prefix).
+
+*You apply a release tag to a specific commit. The [.tekton/push.yaml](.tekton/push.yaml) pipeline
+should have built the image for that commit already. This is the "corresponding image" that receives
+the new version tag. If the image for the tagged commit does not exist, the release pipeline will fail.*
+
+You can watch the release pipeline in the [OpenShift console][ocp-cachi2-pipelines] in case it fails
+(the pipeline is not visible anywhere in GitHub UI). For intermittent failures, retrying should be
+possible from the OpenShift UI or by deleting and re-pushing the version tag.
+
+*⚠ The release pipeline runs as soon as you push a tag into the repository. Do not push the new version
+tag until you are ready to publish the release. You can use GitHub's ability to auto-create the tag
+upon publishment.*
+
+## Release schedule
+
+Cachi2 has a bi-monthly release schedule. Note that the release might be skipped if there's no significant changes to release.
```

Comment:
Not sure if these belong here or in the main README

Reply:
![image](https://github.com/user-attachments/assets/648e716d-88be-4369-b642-3e1fbf15638e)

Thanks, English

Reply:
In my defense, my brain was almost shutting down when I wrote this. I'll change it to "a release every fortnight", to make it clearer.

Reply:
How about just "This project follows a two week release cycle" instead?

Reply:
For sure, the fortnight had a heavy /s implied :laughing:

Reply:
Fixed.

Reply:
I was really looking forward to the influx of Gen-Z contributors thinking we were some kind of Fortnite project :stuck_out_tongue: